src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/EmailAddressAttribute.cs

```diff
@@ -32,7 +25,10 @@ public override bool IsValid(object value)
             }
 
             var valueAsString = value as string;
-            return valueAsString != null && _regex.Match(valueAsString).Length > 0;
+            return (valueAsString != null
+                && valueAsString.Count(c => c == '@') == 1
+                && valueAsString[0] != '@'
+                && valueAsString[valueAsString.Length-1] != '@');
```

Comment:
Could we avoid using LINQ here?  I don't know how often IsInvalid is called, so maybe it doesn't matter, but the Count call is going to cause an allocation and use interface calls when it's not really necessary.  This block could be reimplemented something like:
```C#
if (valueAsString == null)
{
    return false;
}

bool found = false;
for (int i = 0; i < valueAsString.Length; i++)
{
    if (valueAsString[i] == '@')
    {
        if (found || i == 0 || i == valueAsString.Length - 1)
        {
            return false;
        }
        found = true;
    }
    return found;
}
```